### ISSUE TITLE
Shortcut not registering when 

### ISSUE THREAD

**Issue by odajay** [Cito/Pinboard-Pin#14]
I edited my userChrome.css to create a borderless Firefox to make more like [Qutebrowser](https://qutebrowser.org) for practical and aesthetic reasons, as I mainly use keyboard shortcuts. The instructions were taken from the Firefox forum [here](https://support.mozilla.org/en-US/questions/1216436)
It seems that because of that, the extension is not triggered when I `ALT + P`.
I use now this different extension which works by [creating a pop-up](https://addons.mozilla.org/en-US/firefox/addon/pinboardin/?src=external-bn). I miss the additional functionalities of Pinboard-Pin. I realize though it's an extreme edge case. 

Is it impossible to fix this behaviour?

**Comment by Cito** [comment]
Are you sure you only added the userChrome.css for the scroll bar? Because that should really not interfere with Pinboard-Pin. Just tried it, and found no problem - the keyboard shortcut works as well. You must have made another change or installed a different add-on that snatched the keyboard shortcut.

**Comment by odajay** [comment]
Here is the full userChrome.css I implemented:

```
@namespace url("http://www.mozilla.org/keymaster/gatekeeper/there.is.only.xul"); /* set default namespace to XUL */

/*
 * Hide tab bar, navigation bar and scrollbars
 * !important may be added to force override, but not necessary
 */π
 #TabsToolbar {visibility: collapse;}
 #navigator-toolbox {visibility: collapse;}
 :-moz-any(#content,#appcontent) browser{
    margin-right: -14px !important;
    /* margin-bottom: -14px !important; */
    overflow-y: scroll !important;
    overflow-x: scroll !important;
 }
```
I just tested and the shortcut doesn't work with Pinboard-Pin. The thing is that the shortcut works with [this other addon](https://addons.mozilla.org/en-US/firefox/addon/pinboardin/?src=external-bn).

**Comment by Cito** [comment]
All right, when I use the CSS that you posted (hiding the `#navigator-toolbox` seems to be the crucial issue here) the Ctrl+P shortcut also stops working for me. But using that CSS I also don't see the address bar anymore and Ctrl+L doesn't show it either. So how do you even use your browser? And the other add-on doesn't work with that CSS either. You still seem to do something differently.

**Comment by odajay** [comment]
I use [Vim-Vixen](https://addons.mozilla.org/en-US/firefox/addon/vim-vixen/?src=search) and [Vertical Tabs Reloaded](https://addons.mozilla.org/en-US/firefox/addon/vertical-tabs-reloaded/?src=search). I can assure that the other addon works flawlessly on my side. 

Here are my other addons if that can help, I don't use that many of them:

- 1password extension
- Add to Dropmark
- DuckDuckGo Privacy Essential
- Grammarly
- New Tab Homepage
- OneTab
- Qbserve
- Reddit for YouTube
- Tampermonkey (with only a script for TT-RSS)
- uBlock Origins

I tried disabling all but Pinboard-Pin but even that didn't work :(

**Comment by Cito** [comment]
I see, you're using that vim addon to open pages. Btw, it's nice, I might use that in the future as well. I was able to reproduce your observation now and make it work with the other addon that does not use a popup.

Regarding the keyboard shortcut, no good news. I did some investigation. The other add-on binds the keyboard shortcut to a custom command and catches the command in a background script. In principle I can do that as well in Pinboard-Pin. However, as far as I see it's currently not possible in Firefox to open the Pinboard-Pin popup as reaction to the command. See [here](https://developer.mozilla.org/en-US/Add-ons/WebExtensions/API/browserAction/openPopup):

> Activating a keyboard shortcut defined by the extension. (Note: this is not currently supported in Firefox.  Instead consider binding the keyboard shortcut to the `_execute_browser_action` special action.)

In fact I'm getting a "browserAction.openPopup may only be called from a user input handler" error when I try to open the popup in the background script. Binding to `_execute_browser_action` is exactly what I'm doing already. But that works only when the popup button is visible, as you experienced.

If you have any other idea how to solve it, let me know. I'm closing this for now.
